modules/gateway/gateway_test.go

```diff
@@ -17,11 +17,6 @@ func newTestingGateway(name string, t *testing.T) *Gateway {
 	if err != nil {
 		t.Fatal(err)
 	}
-	// Manually add myAddr as a node. This is necessary because g.addNode
-	// rejects loopback addresses.
-	id := g.mu.Lock()
-	g.nodes[g.myAddr] = struct{}{}
-	g.mu.Unlock(id)
```

Comment:
makes me think: wherever we have a comment like: "this is necessary because," it would be ideal to include an assertion that it is indeed the case. In this case, we would assert that `g.nodes` does not already contain `g.myAddr`.